### ISSUE TITLE
Troubleshooting the script to work with 10.10 and 10.11

### ISSUE THREAD

**Issue by seesolve** [seesolve/CreateHtmlEmailSignature#4]
@benthomasian commented in issue #3, regarding trying to get the script to work with 10.10 and 10.11. I am continuing that conversation here, so that other users with a similar issue can find appropriate help easily.

**Comment by seesolve** [comment]
@benthomasian, the script does delete all Mail signatures named "Company Standard" ... Were all your signatures named that?

**Comment by benthomasian** [comment]
No, they for sure were not named that, they had different names. Waiting for my developer buddy to sign and send me the compiled script to test but the Script works fine on it's own if there are no signatures there currently on the users system.

**Comment by seesolve** [comment]
I'm concerned because most people already have at least one signature already on their system and we don't want to mess that up.

**Comment by benthomasian** [comment]
Yeah, in my case I just pulled my signatures back from my sent emails and re-applied them but it's a good point.  Since you don't have 10.10, or 10.11 I am happy to be a test subject.  Maybe I'll create a couple dummy sigs and try it again.

**Comment by seesolve** [comment]
Interesting. Visited [Coneybeare's instructions](http://matt.coneybeare.me/how-to-make-an-html-signature-in-apple-mail-for-el-capitan-os-x-10-dot-11/) and I see that the steps have changed significantly, and so have the file locations. I'll do some more reading and see what I can come up with.

**Comment by seesolve** [comment]
Whether we start a new version or we modify the original code, the code is going to need more than a superficial rewrite (based on what I'm seeing on the step-by-step instructions at the link you provided). 

The script already looks for the most recently created signature to edit.

And there are more problems to script for in the new versions than that. Reading the instructions at the [link](http://matt.coneybeare.me/how-to-make-an-html-signature-in-apple-mail-for-yosemite-os-x-10-dot-10/) and the [link](http://matt.coneybeare.me/how-to-make-an-html-signature-in-apple-mail-for-el-capitan-os-x-10-dot-11/) you provided, is step 2 necessary? If you simply create a signature but do not associate it with any email address and quit Mail, is the signature still there when you restart Mail?

**Comment by benthomasian** [comment]
I'll try this again tomorrow

**Comment by seesolve** [comment]
Sounds good. Thanks for the help!

**Comment by seesolve** [comment]
Ugh. Well, I think locking the file is going to be easier to script than trying to drag and drop the signature to an email account. I think.

When you locked the file, did you quit Mail first, then find the signature file and lock it through Finder? If not, what steps did you do to lock the file?

@benthomasian, actually, there may be another way. What happens if you click the email account you want the signature to be added to instead? Instead of clicking All Signatures, click an email account and then add the Company Standard signature to that, then quit Mail and restart. Is the signature still there?

Also, from my previous experience, there was an issue with iCloud being on and signatures not being saved. For the computer(s) you are testing this on, what's the iCloud status? 

Closing this for now. For anyone trying to follow along with the process, it's going to make it much easier to have everything in one issue. I've added my last comment back at #4, to make it easier to do so. Let's continue the discussion there.

**Comment by benthomasian** [comment]
I created the file in Mail like your placeholder exactly then quit mail.  I then opened the folder referenced in the link from Coneybeare's page and then found the file with the latest time/date stamp and replaced the placeholder text with the HTML code and locked the file. Only after that did I re-open Apple Mail and set the signature to an account by default.

None of my computers utilize iCloud.

**Comment by seesolve** [comment]
> None of my computers utilize iCloud signatures.
That's good. It'll make it easier to code the basics for now.

Try using this code snippet tomorrow and let me know if you have any errors, and if the signature stays without you needing to lock the file.

```
set textButtonOkay to "Okay"
set textButtonCancel to "Cancel"


tell application "Mail"
	activate
	-- we only need this delay for testing purposes
	delay 1.2
	
	set listEmailAccounts to name of every account
	
	
	set textEmailAccount to item 1 of (choose from list listEmailAccounts with title "Add Signature To Email Account" with prompt "Choose one email account to which the company signature will be added." default items {} OK button name textButtonOkay cancel button name textButtonCancel without multiple selections allowed and empty selection allowed)
end tell


tell application "System Events"
	tell application process "Mail"
		keystroke "," using command down
		delay 1.2
		
		click button "Signatures" of toolbar 1 of window 1
		delay 1.2
		
		-- remember to change this to a try statement in case anything goes wrong.
		tell window "Signatures"
			set uiElementEmailAcctRow to item 1 of (every row of table 1 of scroll area 1 of splitter group 1 of group 1 of group 1 whose value of static text 1 = textEmailAccount)
			delay 0.5
			
			select uiElementEmailAcctRow
			delay 0.5
			
			click button 1 of group 1 of splitter group 1 of group 1 of group 1
			delay 1.2
			
			keystroke "Company Standard"
			delay 0.5
			
			keystroke return
			delay 1.2
			
			-- the 'click at' command refused to work (aargh), so we're changing the focus to enter placeholder text into the signature, so a signature file is created in Finder
			tell scroll area 3 of splitter group 1 of group 1 of group 1 to set focused to true
			delay 1.2
			
			keystroke "Placeholder"
			delay 1.2
			
			keystroke "w" using command down
			delay 3
		end tell
	end tell
end tell


tell application "Mail" to quit
```

**Comment by benthomasian** [comment]
That script above worked perfectly so far! In the email account that I want to create the signature in there is now a "Placeholder" signature for the account I selected but that's not where the issue occurs.  It happens later when Apple Mail is re-opened by the script and the text is replaced in an existing signature.

**Comment by seesolve** [comment]
Yes, we're taking care of this step by step, to build this properly. The reason it's gibberish is because now we have to open up the signature and add the html without erasing the metadata already in the signature.

I won't be able to work on this again 'til Sunday.

**Comment by seesolve** [comment]
Well. Here's a little something until Sunday. Try the following only on a 10.11 computer, that already has a Company Standard signature created with 'Placeholder' as its text. (And the Company Standard signature has to be the most recent signature created.) I want to know if this succeeds in changing the text of the signature without producing gibberish.

```
-- >> orig script does not include w/o folder creation. correct this.
set textUserLibraryFolderPath to (path to library folder from user domain without folder creation) as text

set textMailSignaturesElCapFolderPath to textUserLibraryFolderPath & "Mail:V3:MailData:Signatures:"

set textMailSignatureFileNameExtension to "mailsignature"


tell application "Finder"
	set listMailSignatures to every file of alias textMailSignaturesElCapFolderPath whose name ends with textMailSignatureFileNameExtension
	
	(*
	if exists alias textMailSignaturesIcloudFolderPath then
		set listMailSignatures to (every file of alias textMailSignaturesIcloudFolderPath whose name ends with textMailSignatureFileNameExtension)
	end if
	*)
	
	-- >> orig script has this as item 1. correct this.
	set aliasMailSignature to item -1 of (sort listMailSignatures by creation date) as alias
	
	(*
	if boolVersionLion is true then
		set textMailSignaturePosixPath to POSIX path of aliasMailSignature
	end if
	*)
end tell

-- >> remember to add a try/error block here
open for access aliasMailSignature with write permission
set textMailSignature to read aliasMailSignature
set numMetadataOffset to offset of (linefeed & linefeed) in textMailSignature

set eof of aliasMailSignature to (numMetadataOffset + 1)

write "booyaka" to aliasMailSignature starting at eof as text
close access aliasMailSignature
```
